src/ripple/core/JobCounter.h

```diff
@@ -159,8 +155,7 @@ class JobCounter
         // construct the wrapped job if we were not previously joined.
         // That would introduce a timing window.
         boost::optional<CountedJob<JobHandler>> ret {
-            CountedJob<JobHandler> {
-                *this, std::forward<JobHandler> (jobHandler)}};
+            CountedJob<JobHandler> {*this, std::move (handler)}};
```

Comment:
@miguelportilla I think I'll be able to use this suggestion once I've implemented your suggestion of using `operator++` to increment the count.  Then I won't need to lock `mutex_` in order to construct the `JobHandler`.